src/v/cloud_storage/remote_partition.cc

```diff
@@ -433,12 +445,62 @@ class partition_record_batch_reader_impl final
         }
     }
 
+    ss::future<> init_cursor(storage::log_reader_config config) {
+        int retry_quota = 4;
+        if (config.first_timestamp.has_value()) {
+            // Find manifest that contains requested timestamp
+            while (true) {
+                auto cur = co_await _partition->_manifest_view->get_active(
+                  config.first_timestamp.value());
+                if (cur.has_failure()) {
+                    if (
+                      cur.error() == error_outcome::repeat
+                      && retry_quota-- > 0) {
+                        continue;
+                    }
+                    vlog(
+                      _ctxlog.error,
+                      "Failed to query spillover manifests by time: {}",
+                      cur.error());
+                    throw std::system_error(cur.error());
+                }
+                _view_cursor = std::move(cur.value());
+                break;
+            }
+        } else {
+            while (true) {
+                // NOTE: config.start_offset actually contains kafka offset
+                // stored using model::offset type.
+                auto res = co_await _partition->_manifest_view->get_active(
+                  model::offset_cast(config.start_offset));
+                if (res.has_failure()) {
+                    if (
+                      res.error() == error_outcome::repeat
+                      && retry_quota-- > 0) {
+                        continue;
+                    }
+                    vlog(
+                      _ctxlog.error,
+                      "Can't fetch spillover manifest, error: {}",
+                      res.error());
+                    co_return;
+                }
+                _view_cursor = std::move(res.value());
+                break;
+            }
+        }
```

Comment:
You could init the `async_view_search_query_t` variant before the while loop and remove the duplicated code. Something like:
```
async_view_search_query_t query;
if(config.first_timestamp.has_value()) {
  query = config.first_timestamp.has_value();
} else {
  query = model::offset_cast(config.start_offset);
}

while (true) { ... }
```

Reply:
done